packages/protocol/contracts/L1/libs/LibTokenomics.sol

```diff
@@ -53,107 +56,100 @@ library LibTokenomics {
+        TaikoData.Config memory config,
+        uint64 proofTime
     )
         internal
         view
-        returns (uint64 newFeeBase, uint64 reward, uint64 premiumRate)
+        returns (uint64 reward, uint64 newProofTimeIssued, uint64 newBasefee)
     {
-        if (proposedAt > provenAt) revert L1_INVALID_PARAM();
-
-        (newFeeBase, premiumRate) = getTimeAdjustedFee({
-            feeConfig: config.provingConfig,
-            feeBase: state.feeBase,
-            isProposal: false,
-            timeUsed: provenAt - proposedAt,
-            timeAverage: state.avgProofTime
-        });
-        reward = getSlotsAdjustedFee({
-            state: state,
-            config: config,
-            isProposal: false,
-            feeBase: newFeeBase
-        });
+        newProofTimeIssued = state.proofTimeIssued;
 
-        unchecked {
-            reward = uint64((reward * (10000 - config.rewardBurnBips)) / 10000);
-        }
-    }
+        newProofTimeIssued = (newProofTimeIssued > config.proofTimeTarget)
+            ? newProofTimeIssued - config.proofTimeTarget
+            : uint64(0);
+        newProofTimeIssued += proofTime;
 
-    // Implement "Slot-availability Multipliers", see the whitepaper.
-    function getSlotsAdjustedFee(
-        TaikoData.State storage state,
-        TaikoData.Config memory config,
-        bool isProposal,
-        uint64 feeBase
-    ) internal view returns (uint64) {
-        unchecked {
-            // m is the `n'` in the whitepaper
-            uint256 m = 1000 *
-                config.maxNumProposedBlocks +
-                config.slotSmoothingFactor;
-            // n is the number of unverified blocks
-            uint256 n = 1000 *
-                (state.numBlocks - state.lastVerifiedBlockId - 1);
-            // k is `m − n + 1` or `m − n - 1`in the whitepaper
-            uint256 k = isProposal ? m - n - 1000 : m - n + 1000;
-            return uint64((feeBase * (m - 1000) * m) / (m - n) / k);
+        newBasefee = _calcBasefee(
+            newProofTimeIssued,
+            config.proofTimeTarget,
+            config.adjustmentQuotient
+        );
+
+        uint64 numBlocksBeingProven = state.numBlocks -
+            state.lastVerifiedBlockId -
+            1;
+        if (numBlocksBeingProven == 0) {
+            reward = uint64(0);
+        } else {
+            uint64 totalNumProvingSeconds = uint64(
+                uint256(numBlocksBeingProven) *
+                    block.timestamp -
+                    state.accProposedAt
+            );
+            ///@dev If block timestamp is equal to state.accProposedAt (not really, but theoretically possible)
+            ///@dev there will be division by 0 error
+            if (totalNumProvingSeconds == 0) {
+                totalNumProvingSeconds = 1;
+            }
+
+            reward = uint64(
+                (uint256(state.accBlockFees) * proofTime) /
+                    totalNumProvingSeconds
```

Comment:
seems that there is a chance that when there are more and more unproven blocks on chain, reward will be less and less? or i missed something?

Reply:
Theoretically possible, in case proposers stop proposing.
The fee connected with the proof time issued seconds, so the more 'unproven' -> the more the fee -> the bigger the reward pool.